shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profilePreferences/presentation/editAddress/components/EditAddressScreen.kt

```diff
@@ -0,0 +1,164 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.vml.tutorial.plantshop.MR
+import com.vml.tutorial.plantshop.core.presentation.UiText
+import com.vml.tutorial.plantshop.core.presentation.asString
+import com.vml.tutorial.plantshop.profilePreferences.presentation.ScreenTitle
+import com.vml.tutorial.plantshop.profilePreferences.presentation.UserInput
+import com.vml.tutorial.plantshop.profilePreferences.presentation.editAddress.EditAddressState
+import com.vml.tutorial.plantshop.ui.theme.Typography
+
+@Composable
+fun EditAddressScreen(state: EditAddressState, onEvent: (EditAddressEvent) -> Unit) {
+    LazyColumn {
+        item {
+            IconButton(
+                modifier = Modifier.padding(8.dp).size(50.dp),
+                onClick = {
+                    onEvent(EditAddressEvent.NavigateBack)
+                }
+            ) {
+                Icon(
+                    imageVector = Icons.Rounded.ArrowBack,
+                    contentDescription = null,
+                    tint = MaterialTheme.colorScheme.onPrimaryContainer
+                )
+            }
+        }
+        item {
+            Column(
+                modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp),
+                verticalArrangement = Arrangement.spacedBy(16.dp)
+            ) {
+                ScreenTitle(
+                    UiText.StringRes(MR.strings.edit_address_title_text).asString(),
+                    UiText.StringRes(MR.strings.edit_address_subtitle_text).asString(),
+                    Modifier.padding(horizontal = 8.dp)
+                )
+
+                Spacer(modifier = Modifier.height(16.dp))
+
+                UserInput(
+                    state = state.streetName,
+                    placeholderText = UiText.StringRes(MR.strings.edit_address_street_name_placeholder_text)
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Text, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(EditAddressEvent.StreetNameChanged(it))
+                }
+
+                UserInput(
+                    state = state.doorNumber?.toString().orEmpty(),
+                    placeholderText = UiText.StringRes(MR.strings.edit_address_door_number_placeholder_text)
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Number, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(EditAddressEvent.DoorNumberChanged(it.toInt()))
```

Comment:
This will crash if the field is empty -> "".toInt(). Please check it if it is empty or not.

Reply:
You are right. Refactored